ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift

```diff
@@ -586,12 +618,44 @@ private extension CommonFunctionality {
         return Self.createErrorContainer(error: error, userCancelled: userCancelled)
     }
 
+    static func toPresentedOfferingContext(presentedOfferingContext: [String: Any]?) -> PresentedOfferingContext? {
+        guard let presentedOfferingContext, let offeringIdentifier = presentedOfferingContext["offeringIdentifier"] as? String else {
+            return nil
+        }
+
+        let placementIdentifier = presentedOfferingContext["placementIdentifier"] as? String
+
+        let targetingContext: PresentedOfferingContext.TargetingContext?
+        if let revision = presentedOfferingContext["targetingRevision"] as? Int, let ruleId = presentedOfferingContext["targetingRuleId"] as? String {
+            targetingContext = .init(revision: revision, ruleId: ruleId)
+        } else {
+            targetingContext = nil
+        }
+
+        return PresentedOfferingContext(
+            offeringIdentifier: offeringIdentifier,
+            placementIdentifier: placementIdentifier,
+            targetingContext: targetingContext
+        )
+    }
+
     static func package(withIdentifier packageIdentifier: String,
                         offeringIdentifier: String,
+                        presentedOfferingContext: PresentedOfferingContext?,
                         completion: @escaping(Package?) -> Void) {
         Self.sharedInstance.getOfferings { offerings, error in
             let offering = offerings?.offering(identifier: offeringIdentifier)
-            let package = offering?.package(identifier: packageIdentifier)
+            let foundPackage = offering?.package(identifier: packageIdentifier)
+
+            let package = foundPackage.flatMap { pkg in
+                Package(
```

Comment:
I wonder if we should change the completion block in this function to return a map instead of a `Package` to avoid having to recreate the package here... We can go with this for now though.